include/extractor/scripting_environment_lua.hpp

```diff
@@ -13,12 +13,24 @@
 #include <mutex>
 #include <string>
 
+#include <osmium/index/map/sparse_file_array.hpp>
+#include <osmium/index/map/dense_file_array.hpp>
+#include <osmium/handler/node_locations_for_ways.hpp>
+#include <osmium/dynamic_handler.hpp>
+
 struct lua_State;
 
 namespace osrm
 {
 namespace extractor
 {
+  
+using index_neg_type = osmium::index::map::Dummy<osmium::unsigned_object_id_type, osmium::Location>;
+using index_type = osmium::index::map::SparseFileArray<osmium::unsigned_object_id_type, osmium::Location>;
```

Comment:
Eventually we want a sparse and a dense cache as in

```cpp
using SparseIndex = osmium::index::map::SparseMemArray<osmium::unsigned_object_id_type, osmium::Location>;
using DenseIndex = osmium::index::map::DenseMemArray<osmium::unsigned_object_id_type, osmium::Location>;
```

and either let the user decide which one to use or dispatch based on the base map .pbf's size: dense for planet, sparse for everything considerably smaller.